src/openzeppelin/access/ownable/ownable.cairo

```diff
@@ -1,92 +1,80 @@
-#[contract]
+#[starknet::contract]
 mod Ownable {
     use openzeppelin::access::ownable::interface;
     use starknet::ContractAddress;
     use starknet::get_caller_address;
     use zeroable::Zeroable;
 
+    #[storage]
     struct Storage {
         _owner: ContractAddress
     }
 
     #[event]
-    fn OwnershipTransferred(previous_owner: ContractAddress, new_owner: ContractAddress) {}
-
-    impl OwnableImpl of interface::IOwnable {
-        fn owner() -> ContractAddress {
-            _owner::read()
-        }
-
-        fn transfer_ownership(new_owner: ContractAddress) {
-            assert(!new_owner.is_zero(), 'New owner is the zero address');
-            assert_only_owner();
-            _transfer_ownership(new_owner);
-        }
+    #[derive(Drop, starknet::Event)]
+    enum Event {
+        OwnershipTransferred: OwnershipTransferred
+    }
 
-        fn renounce_ownership() {
-            assert_only_owner();
-            _transfer_ownership(Zeroable::zero());
-        }
+    #[derive(Drop, starknet::Event)]
+    struct OwnershipTransferred {
+        previous_owner: ContractAddress,
+        new_owner: ContractAddress,
     }
 
-    impl OwnableCamelImpl of interface::IOwnableCamel {
-        fn owner() -> ContractAddress {
-            OwnableImpl::owner()
+    #[generate_trait]
+    impl InternalImpl of InternalTrait {
+        fn initializer(ref self: ContractState, owner: ContractAddress) {
+            self._transfer_ownership(owner);
         }
 
-        fn transferOwnership(newOwner: ContractAddress) {
-            OwnableImpl::transfer_ownership(newOwner);
+        fn owner(self: @ContractState) -> ContractAddress {
```

Comment:
Why do we have an internal `owner()` function now?

Reply:
Good call, I had it for testing initially. Will remove